Minor additions to the software framework

mkristien opened the pull request:
Update README with how to lint ignore lines
Add resultant binary to .gitignore
Add compilation flags to Makefile

mkristien committed:
Add compilation flags to Makefile
Update README with how to lint ignore lines
Add resultant binary to .gitignore

mkristien committed:
Git Hooks work

mkristien committed:
Add script for deploying git hooks

brano2 reviewed (commented):
What is the empty `temp` file? Do we need it?

mkristien commented:
Sorry, we do not. I only used it for testing pre-commit hook. I needing some change to make a commit. Will remove the temp file and recommit shortly.

mkristien committed:
Delete temp file

brano2 commented:
Can you also add a note to the README regarding the linter not being 100% accurate and style guide having preference?

mkristien commented:
No, I would prefer first changing the linter to reflect the style guide. Do not approve yet then, we shall resolve the inconsistencies together.

brano2 committed:
Split cleaning and reorder compilation and linting

There is now a separate target to allow for `make clean`ing of linter's
stuff only so that you can re-lint without recompiling.
Also, `make` now first compiles the project and then lints. It is more
important to have code that compiles than code that lints (although the
latter is still super important).

brano2 committed:
Fix opening braces on new line in function definitions

brano2 reviewed (approved):
LGTM

brano2 commented:
Any comments @mkristien?

mkristien commented:
looks good to me
